weave

```diff
@@ -96,15 +96,21 @@ usage() {
 }
 
 docker_sock_volume_mount() {
-  if [ -z "$DOCKER_HOST" ]; then
-      DOCKER_SOCK="/var/run/docker.sock"
-  elif echo "$DOCKER_HOST" | grep -q "^unix://" >/dev/null; then
+  if echo "$DOCKER_HOST" | grep -q "^unix://" >/dev/null; then
       DOCKER_SOCK="${DOCKER_HOST#unix://}"
+  else
+      DOCKER_SOCK="/var/run/docker.sock"
   fi
-  [ -z "$DOCKER_SOCK" ] || echo "-v $DOCKER_SOCK:$DOCKER_SOCK"
+  echo "-v $DOCKER_SOCK:$DOCKER_SOCK"
 }
 
 exec_remote() {
+    # Pass through DOCKER_HOST if it is a Unix socket;
+    # a TCP socket may be secured by TLS, in which case we can't use it
+    if echo "$DOCKER_HOST" | grep -q "^unix://" ; then
```

Comment:
it's unfortunate to have this check in two places....

perhpas `docker_sock_volume_mount` should be renamed and return both the `-v ...` and `-e DOCKER_HOST`, when appropriate.